src/main/java/frc/team88/swerve/gyro/Pigeon.java

```diff
@@ -0,0 +1,55 @@
+package frc.team88.swerve.gyro;
+
+import com.ctre.phoenix.motorcontrol.can.TalonSRX;
+import com.ctre.phoenix.sensors.PigeonIMU;
+
+public class Pigeon implements SwerveGyro {
+
+  // The navx object that this is based on.
+  private PigeonIMU base;
+
+  // The offset to add to the yaw
+  private double offset = 0;
+
+  /** Constructor. Pigeon doesn't really have a default assumes CAN ID 0. */
```

Comment:
We should probably have a constructor that takes in a PigeonIMU object for reasons that I'm going to explain in another comment. I'm realizing I should have done the same for the navX

Reply:
Well, I didn't really explain it in that comment, so briefly: by giving teams the ability to pass a PigeonIMU into the Pigeon constructor, and then a Pigeon into the SwerveController constructor, they can retain access to the original PigeonIMU object instead of having it be locked away behind private fields